h/views/activity.py

```diff
@@ -92,31 +95,76 @@ def group_search(request):
 
     return result
 
+
+@view_config(route_name='activity.user_search',
+             request_method='GET',
+             renderer='h:templates/activity/search.html.jinja2')
+def user_search(request):
+    if not request.feature('search_page'):
+        raise httpexceptions.HTTPNotFound()
+
+    username = request.matchdict['username']
+
+    result = search(request)
+
+    result['opts'] = {'search_username': username}
+    result['more_info'] = 'more_info' in request.params
+
+    user = request.find_service(name='user').fetch(
+        'acct:{username}@{authority}'.format(username=username,
+                                             authority=request.auth_domain))
```

Comment:
This is a bit of technical debt (that some view code should need to know how to construct a userid from a username). We used to have a helper function for this but `fdd39757b62308680ba55c98557abeb1bdf52bb0` removed it. There us a `.userid` property on the `User` model instead. The problem is that the user service does not have a method for fetching a user by username and authority, it requires the userid, so I can't get the user model object without having the userid first, unless I access the model code directly instead of going through the service.

I think the solution is to add a `fetch_by_username(username, authority)` to the user service